### ISSUE TITLE
'RangeError: Maximum call stack size exceeded' When processing html with large number of <wbr> tags

### ISSUE THREAD

**Issue by andrewtamura** [werk85/node-html-to-text#44]
Steps to reproduce:
1. Install `html-to-text` version 1.2.1
2. Create html file with 2000 `<wbr>` tags. See the sample python script
3. Run the command `cat test-bad.html | html-to-text > out.txt`
4. Observe `'RangeError: Maximum call stack size exceeded'`

```
# Python script to generate HTML
f = open('test-bad.html','w')
f.write('<!DOCTYPE html><html><head></head><body>\n')
for n in range(2000):
    f.write('<wbr>n')
f.write('</body></html>')
f.close()
```

This is also reproducible using invalid HTML tags instead of `<wbr>` tags e.g. `<foo>`

**Comment by mlegenhausen** [comment]
Could please post a stack trace?

**Comment by andrewtamura** [comment]
```
/Users/andrew/Code/testing-html-to-text/node_modules/html-to-text/lib/html-to-text.js:133
					elem.needsSpace = whiteSpaceRegex.test(result);
					                                  ^
RangeError: Maximum call stack size exceeded
```

**Comment by bradvogel** [comment]
Bump on this @mlegenhausen :)

**Comment by mlegenhausen** [comment]
Sorry hadn't the time to investigate. I think the problem will be the used HTML parser, which simply hits the max call stack size when parsing the html tags recursively. Could you try to increase node's stack size limit via `node --stack-size=2048 ./bin/cli.js` and give me some feedback?

**Comment by andrewtamura** [comment]
increasing the stack size with `--stack-size=2048` allows a 2000 element html, like I've described above, to be parsed without errors. However creating a bad html file with more elements can trigger the RangeError exception.

**Comment by mlegenhausen** [comment]
I think we need to update or switch the `htmlparser` to solve this problem. I think switching to [htmlparser2](https://github.com/fb55/htmlparser2) would make more sense, cause the htmlparser project seems to make no further progress.

Every pull request is welcome ;)

**Comment by mlegenhausen** [comment]
Fixed by #44
